### DISCUSSION TITLE
SmartAutoClicker-1.4.0

### DISCUSSION THREAD

**Discussion by Nain57** [Nain57/Smart-AutoClicker#125] [Announcements]
# New Features:

## Intent actions
You can now create a new type of action of your events, an Intent. An Intent is the Android way to communicate between applications. With this, you can start another app easily. UI is adapted for simple and advanced users. Broadcast are also supported !

## Detection Quality
You can now change the quality of the detection in the new scenario settings screen. For a faster detection, but with more false positive, move the cursor to "Speed". For a slower detection, but with an higher detection precision, move the cursor to "Precision". This setting is unique to each scenario, so you can adapt it to your needs. Don't forget to use the debugger to find the best setting.

## End condition / Stop after refactoring:
It was clear that this feature wasn't understood by everybody. Yet, it was really useful so I decided to display it differently. It is now located in the new scenario settings screen. Here, you can see all the end conditions, and even provide a operator like the event's conditions.

## New toolbar
As the application grows in features, the button bar of the event list dialog is becoming crowed. To fix this, the buttons are now displayed in icons, and shown only when needed. This new bar is also the host of the new scenario settings button, providing access to the previous new features.

# Bug fixes:

## Bluestacks compatibility
Smart AutoClicker is now compatible with Bluestacks (tested on version 5.7.100.1036). I had to change the "graphic engine mode" to "compatibility" to run the application but this might be my old computer.

## Copy button not displayed
When creating a new event, if the list of action/conditions was empty, the copy button was not displayed. Same for the event list. This was hiding the copy from other scenario feature. This is now fixed.
 

<hr /><em>This discussion was created from the release <a href='https://github.com/Nain57/Smart-AutoClicker/releases/tag/1.4.0'>SmartAutoClicker-1.4.0</a>.</em>

**Comment by Tsyoee** [comment]
it's bug sir,,,I entered a pink color capture and it will auto-click when the color appears, however, it auto-clicks before the pink target appears.,, please fix it sir,, i'm using bluestack 5.7.100.1036 N32,, iff u can make 1.2.4 version suport for bluestack,, it's perfect,,

**Reply by Nain57** [reply]
I have replied in your bug request: https://github.com/Nain57/Smart-AutoClicker/issues/126

I need more information, but it seems more like a scenario configuration issue. I will gladly help in your issue ticket.

**Comment by xTeooo** [comment]
i can confirm that it works with BS v4 and LDPlayer 4

**Comment by MrKidney** [comment]
The version 1.4.0 can't run correctly on redmi k30s(mi 10t). The system on my phone is pixelexperience 12.1.
When starting a scenario, the menu will disappear immediately. Then I tried to create a new scenario, but the screenshot can't run correctly when the phone is in landscape mode(screenshot is attached
![Uploading Screenshot_20220519-134151_命运-冠位指定.png…]()
) If I swipe the screenshot to a correct position, then add press event, the position to press would be wrong.

**Reply by MrKidney** [reply]
Feel sorry for my poor English. And I'll be  delight if you can fix the bug.🌹
I can  provide more things(such as the log) if needed.

**Reply by Nain57** [reply]
Hi and thanks for your feedback !
This seems related to this issue: https://github.com/Nain57/Smart-AutoClicker/issues/128

I'm aware of it and it should be fixed in the next version.

**Comment by ssneoh** [comment]
Hello. Can we make the image recognition to resize to a small square size instead of the default rectangle (smallest size)?
I'm unable to detect my image because I need only a small area of the image to be detected.

Kindly advice & thank you .

**Reply by jd0g-23** [reply]
Do you mean even smaller than after you resized it to the smallest option? If you drag the edges of the box it is resizable to a pretty small size. Sorry, maybe I'm not understanding from where you say "instead of the default rectangle (smallest size)?".

**Reply by ssneoh** [reply]
Hi. For this, I've attached 2 images. I've used app "Macrorify" as reference.
As you see, when I try to capture the image & resize it to minimum, SmartAutoClicker will make longer image.

Hope you understand what I mean because my English is not good. Really wanted to use SmartAutoClicker because it's free (thanks!) and more user friendly.

![macrorify_capture_small](https://user-images.githubusercontent.com/1488066/170488529-3d84e26d-0f6b-4e86-98d3-8a383acd1124.png)

![smartautoclicker_capture_small](https://user-images.githubusercontent.com/1488066/170488537-323e177f-2879-4139-b077-93905289b0ab.png)
.

**Reply by Nain57** [reply]
This seems not zoomed, it can seems obvious but did you know that you can pinch to zoom on the capture and create smaller conditions ?

**Reply by ssneoh** [reply]
Ahhh ~ Zooming! Zooming solved the problem.
Many many thanks ;-)

**Reply by Nain57** [reply]
I definitely have to add an hint for this zoom feature, I feel like a lot of users are missing it 😅

**Comment by jd0g-23** [comment]
Hello,
I can make mine a box this small...


![Screenshot_20220526-214838_One UI Home](https://user-images.githubusercontent.com/106185920/170619759-8ea11184-96b5-455b-b9c1-a7c13186ebf6.png)

Not sure why you can't square yours off. Maybe a bug with different android versions or implimentations?

Also I don't know if screen resolution matters, as you can see my smallest box size seems bigger compared to the icons size...perhaps we have different icon sizes too.

At any rate, I don't know enough about android internals or the smartclicker to even know where to check in the source code. Maybe it works in version 1.5.0?

I wish I could help more.

Cheers!
